core/src/main/java/hudson/model/Job.java

```diff
@@ -674,7 +674,7 @@ public void renameTo(String newName) throws IOException {
 
     @Override
     public void movedTo(DirectlyModifiableTopLevelItemGroup destination, AbstractItem newItem, File destDir) throws IOException {
-        Job newJob = (Job) newItem; // Missing covariant parameters type here.
```

Comment:
I thought this comment was related to removed line. I guess it used to check types even if it not used.